serving/src/main/java/feast/serving/util/RefUtil.java

```diff
@@ -21,7 +21,13 @@
 
 public class RefUtil {
   public static String generateFeatureStringRef(FeatureReference featureReference) {
```

Comment:
I think we can do away with these string references and straight up use `FeatureSetReference`s and `FeatureReference`s in the maps.

Reply:
Yeah I think its a good idea. Although I think we should address in a separate dedicated techdebt/refactoring PR. Related issue: #674